api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -226,8 +211,33 @@ public void setNextSequenceValue(Long nextSequenceValue) {
 		this.nextSequenceValue = nextSequenceValue;
 	}
 	
-	private PrefixProvider getPrefixProvider(String prefixProviderBean) {
-		return Context.getRegisteredComponent(prefixProviderBean, PrefixProvider.class);
+	/**
+	 * Gets the {@link PrefixProvider} from a configured {@code prefix}
+	 * 
+	 * @should return null if a static prefix is provided
+	 * @should return null if the configured provider is not found
+	 * @param prefix
+	 * @return the PrefixProvider
+	 */
+	public PrefixProvider getPrefixProvider(String prefix) {
+		if (StringUtils.isBlank(prefix)) {
+			return null;
+		}
+		if (prefix.startsWith(CONFIGURATION_PREFIX)) {
+			String prefixProviderBean = "";
+			try {
+				prefixProviderBean = prefix.substring(prefix.indexOf(":") + 1).trim();
+				return Context.getRegisteredComponent(prefixProviderBean, PrefixProvider.class);
+			}
+			catch (IndexOutOfBoundsException e) {
+				log.error("Invalid Prefix Configuration. PrefixProvider bean name not provided", e);
+			}
+			catch (APIException e) {
```

Comment:
Why `APIException`, wouldn't you come across some Spring bean error instead?

Reply:
Then you may as well catch _any_ other exception:
```java
catch (Exception e)
```

Reply:
> Why `APIException`, wouldn't you come across some Spring bean error instead?

The error is thrown as an `APIException`. No spring bean error.

Reply:
I would just catch all exceptions in one go:
```java
catch (Exception e) {
  log.error(
      "Invalid prefix configuration. The PrefixProvider bean class or bean name could not be parsed. Here is an example of a valid configuration: '"
              + CONFIGURATION_PREFIX + LocationBasedPrefixProvider.class.getSimpleName() + "'.",
      e);
}
```